web/modules/custom/asset_distribution/src/Controller/AssetDistributionController.php

```diff
@@ -48,18 +74,14 @@ public function add(RdfInterface $rdf_entity) {
    *   The access result object.
    */
   public function createAssetDistributionAccess(RdfInterface $rdf_entity) {
-    // Check that the passed in RDF entity is a solution, and that the user
-    // has the permission to create distribution entities.
-    // @todo Collection owners and facilitators should also have the right to
-    //   create distributions related to the solutions they manage.
-    // @see https://webgate.ec.europa.eu/CITnet/jira/browse/ISAICP-2450
-    if ($rdf_entity->bundle() == 'asset_release' && $this->currentUser()
-        ->hasPermission('create asset_distribution rdf entity')
-    ) {
-      return AccessResult::allowed();
+    $solution = $this->assetDistributionRelations->getReleaseSolution($rdf_entity);
+    $user = $this->currentUser();
+    if (empty($solution) && !$user->isAnonymous()) {
+      return AccessResult::neutral();
     }
```

Comment:
I think I was thinking that I should block it without having to return forbidden but you are right! it should be only forbidden. This form has only one intended use.

The <code>AccessResult::neutral()</code> is the issue here.